src/LinuxTracing/LibunwindstackMaps.cpp

```diff
@@ -13,7 +13,7 @@ class LibunwindstackMapsImpl : public LibunwindstackMaps {
   explicit LibunwindstackMapsImpl(std::unique_ptr<unwindstack::BufferMaps> maps)
       : maps_{std::move(maps)} {}
 
-  unwindstack::MapInfo* Find(uint64_t pc) override { return maps_->Find(pc); }
+  unwindstack::MapInfo* Find(uint64_t pc) override { return maps_->Find(pc).get(); }
```

Comment:
This is a little bit of a contract violation. I think we shouldn't assume that the shared_ptr that is returned by `Find` continues to exist after the function gets out of scope. Would it be possible to return the shared_ptr instead of a raw pointer here?

Reply:
You're right. `shared_ptr` is now returned.